shell/platform/darwin/ios/ios_external_texture_gl_share_context.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_IOS_EXTERNAL_TEXTURE_SHARE_CONTEXT_H_
+#define FLUTTER_SHELL_PLATFORM_IOS_EXTERNAL_TEXTURE_SHARE_CONTEXT_H_
+
+#include "flutter/flow/texture.h"
+#include "flutter/fml/platform/darwin/cf_utils.h"
+#include "flutter/shell/platform/darwin/ios/framework/Headers/FlutterTexture.h"
+
+namespace flutter {
+// This is another solution for Flutter's external texture.
+// Unlike the original external texture which copies pixel buffer from an object that implements
+// FlutterTexture protocol, and creates a texture reference using the pixel buffer, this solution
+// will copy the OpenGL texture directly from the object implementing FlutterShareTexture protocol.
+class IOSExternalTextureShareContext : public flutter::Texture {
```

Comment:
This should probably be named IOSExternalShareContextTexture.  The type is usually placed at the end of the name.